db/errors.go

```diff
@@ -0,0 +1,35 @@
+package db
+
+import (
+	"github.com/go-sql-driver/mysql"
+	"github.com/mattn/go-sqlite3"
+)
+
+func IsDeadlockError(err error) bool {
+	switch t := err.(type) {
+	case *mysql.MySQLError:
+		switch t.Number {
+		case 1213:
+			return true
+		}
+	}
+
+	return false
+}
+
+func IsDuplicateEntryError(err error) bool {
+	switch t := err.(type) {
+	case *mysql.MySQLError:
+		switch t.Number {
+		case 1062:
+			return true
+		}
+	case sqlite3.Error:
```

Comment:
*mysql.MySQLError vs. sqlite3.Error: why do sometimes use pointer, and sometimes use a value?